### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by ellert**
Here are some fixes for failing tests.

### CODE REVIEW

File: math/mathcore/inc/Fit/FitData.h [hunk 1]
```diff
@@ -185,7 +185,7 @@ namespace ROOT {
 
             for (unsigned int i = 0; i < fDim; i++) {
                fCoords[i].resize(fMaxPoints + VectorPadding(fMaxPoints));
-               fCoordsPtr[i] = &fCoords[i].front();
+               fCoordsPtr[i] = fCoords[i].empty() ? NULL : &fCoords[i].front();
```
Comment (pcanal) [inline math/mathcore/inc/Fit/FitData.h]: please use nullptr ;)

Comment (ellert) [reply math/mathcore/inc/Fit/FitData.h]: There where already a lot of NULL in those files and I didn't want to introduce inconsistencies by having both NULL and nullptr in the same file. I have made a commit that replaces all NULL with nullptr in the files changed by the PR.

**Comment by Axel-Naumann** [comment]
@phsft-bot build

**Comment by ellert** [comment]
I rebased the PR because some of the changes proposed had already been applied by other commits.

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by pcanal** [comment]
thanks.

**Pull request merged**